src/strategies/LidoLevL2.sol

```diff
@@ -0,0 +1,235 @@
+        /* Deposit flow */
+        // Trade wEth for wstETH (or equivalent, e.g. cbETH)
+        WETH.safeApprove(address(CURVE), type(uint256).max);
+
+        // Deposit wstETH in AAVE
+        WSTETH.safeApprove(address(AAVE), type(uint256).max);
+
+        // Withdrawals => Trade wstETH for ETH
+        WSTETH.safeApprove(address(CURVE), type(uint256).max);
+        WETH.safeApprove(address(AAVE), type(uint256).max);
+
+        aToken = ERC20(AAVE.getReserveData(address(WSTETH)).aTokenAddress);
+        debtToken = ERC20(AAVE.getReserveData(address(WETH)).variableDebtTokenAddress);
+        AAVE.setUserEMode(1); // 1 = enabled
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                              FLASH LOANS
+    //////////////////////////////////////////////////////////////*/
+
+    IBalancerVault public constant BALANCER = IBalancerVault(0xBA12222222228d8Ba445958a75a0704d566BF2C8);
+
+    enum LoanType {
+        invest,
+        divest
+    }
+
+    function receiveFlashLoan(
+        ERC20[] memory, /* tokens */
+        uint256[] memory amounts,
+        uint256[] memory, /* feeAmounts */
+        bytes memory userData
+    ) external override {
+        require(msg.sender == address(BALANCER), "Staking: only balancer");
+
+        uint256 ethBorrowed = amounts[0];
+        (LoanType loan) = abi.decode(userData, (LoanType));
+
+        if (loan == LoanType.divest) {
+            _endPosition(ethBorrowed);
+        } else {
+            _addToPosition(ethBorrowed);
+        }
+
+        // Payback wETH loan
+        WETH.safeTransfer(address(BALANCER), ethBorrowed);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                         INVESTMENT/DIVESTMENT
+    //////////////////////////////////////////////////////////////*/
+
+    function _afterInvest(uint256 amount) internal override {
+        ERC20[] memory tokens = new ERC20[](1);
+        tokens[0] = WETH;
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = amount.mulDivUp(leverage, 100);
+        BALANCER.flashLoan({
+            recipient: IFlashLoanRecipient(address(this)),
+            tokens: tokens,
+            amounts: amounts,
+            userData: abi.encode(LoanType.invest)
+        });
+    }
+
+    function _addToPosition(uint256 ethBorrowed) internal {
+        // Trade ETHto wstETH
+        uint256 expectedWstETh = _ethToWstEth(ethBorrowed);
+        uint256 wstEth =
+            CURVE.exchange({x: uint256(0), y: 1, dx: ethBorrowed, min_dy: expectedWstETh.slippageDown(slippageBps)});
+
+        // Deposit wstETH in AAVE
+        asset.safeApprove(address(AAVE), type(uint256).max);
+        AAVE.deposit(address(WSTETH), wstEth, address(this), 0);
+
+        // Borrow 90% of wstETH value in ETH using e-mode
+        uint256 ethToBorrow = _wstEthToEth(wstEth).mulDivDown(8999, 10_000);
```

Comment:
Do we need to borrow fixed ratio? 
We can just borrow required amount like `ethBorrowed - asset.balanceOf(this)`. Won't have any idle assets.

Reply:
Great comment! We must always borrow exactly enough to repay the flashloan. Fixed.